### DISCUSSION TITLE
Unable to hear sound, RPi never stops blinking

### DISCUSSION THREAD

**Discussion by iwtwwc** [cardonabits/haxo-hw#73] [Q&A] [answered]
Thanks for building this cool instrument, I want to get mine working! My weak area is Rasp Pi and I bet that's the problem. I downloaded the haxo image and Rasp Pi Installer, used an old iMac with an SD card slot and a 32 GB Sandisk. BTW I had to use the Mac's Disk Utility to erase the disk AND use RPi Imager to FAT32 format the card. And the Mac had to be n the net so I could choose my Pi Zero W 1.1 as device. The writing and verification went fine. I plug the Pi's outer (power) USB into a cord to 1) a USB hub, or 2) a plug-in 5V 2A adaptor. No sound from the 1/8 jack on the Haxophone and the Pi's LED never stops blinking, well it does, and when I try blowing and key-pressing it starts blinking again but no sound. I've tried earbuds and a 1/8 jack to my audio interface (verified that was working with an iPod). What next? I can hook my Mac mini to the Pi's USB OTG and use a serial connection (I'm Arduino experienced but not Pi). Please help, I want to hear this, then use USB, MIDI, etc.

**Comment by iwtwwc** [comment]
Also I apologize for posting this in General, it is definitely Q&A

**Comment by jcard0na** [comment]
Hi @iwtwwc,

Let's get you playing!  If you use RPi Imager, you don't need to be connected to the Internet, nor format the card in advance.  Just pick Operating System -> Custom Image, select the image you downloaded from [https://bit.ly/haxo-rpiz](https://bit.ly/haxo-rpiz) then your SD Card, decline any customizations, and you should be good to go.

![Screenshot 2024-01-14 at 23 50 50](https://github.com/cardonabits/haxo-hw/assets/676181/82683a1c-dad6-4b7c-a052-b68fef445068)

If you hear no sound, you could try to [log into the haxophone](https://github.com/cardonabits/haxo-hw/blob/main/docs/login.md) to see the logs (`journalctl -u haxo`) and check for errors.

Let me know if that helps,

**Reply by WizOfWas** [reply]
I also had this problem.  To diagnose I plugged the sd card into a spare Raspberry Pi that I had a screen/monitor on so I could see the logs.  For me, the issue was a boot loop-- when I used the RaspberryPi imager I has specified "custom settings" and changed the default user name and added my home wifi.  However, I think because the image is read only(? not really sure), it could not change the directory name of the home directory and then eventually it would just reboot and the same thing happened, continuously rebooting.

I am re-imaging the card with this exact process(going onto raspberry pi 3 model B V 1.2, step 6 below is the critical step):
1) Plug in SD-card to computer
2) start Raspberry Pi imager
3) Select "Raspberry Pi Device" = Raspberry Pi 3
4) Select "Operating System" selcted "Use custom " and pick the file "haxophone-rpi3.img" from computer
5) Select "Choose Storage" select the SD-Card
 Click "Next"
**6) to question popup "Would you like to apply OS customisation settings? answer "No, Clear Settings".**
7) next popup "All existing data on "SDCard" will be erased. Are you sure you want to conitnue?  Answer: "Yes"
 then it does the write and verify of the card which takes some time.

**Comment by iwtwwc** [comment]
Thanks for a speedy reply! I kept having failure writing to the SD card when raspberry Pi device said [any] in the Imager. When I connected to the net the rasp pi device field populated with choices including my Zero W. I must have tried 10 times to format the SD card with [any], the custom haxo image, and my SD card. But I suspect I messed up on the customizations tab because I set the pi up for SSID login with username and password to get onto my local wifi network. I wonder if that was a mistake. I'll try re-writing the haxo package with no customization in pi imager, thanks for the suggestion!

**Reply by WizOfWas** [reply]
haha - looks like you were doing this at the same time as me!

I re-imaged and now the boot seems to be ok, but still doesn't seem to work in the haxaphone.. will update this when I figure it out.
The logs just end with it at a login.  (I don't have a keyboard so I could login or do anything else).  In the haxaphone both the red and green light end up just staying on.

**Comment by WizOfWas** [comment]
Got it working!
The imaging with no customization was part 1 of the answer, and the bigger part.

I did get a keyboard and logged in and looked at some logs... took me a minute to find the username/password which is pi/haxophone.

The 2nd thing was if the haxophone audio output jack is touching the raspberry pi hdmi (like the case of the jack just against the case of the hdmi port) then it just emits static and no sound.  I put a piece of tape on the hdmi port and then it all started working.  This may only be a problem with the full-sized raspberry pi-- I am using a the raspberry pi 3.

Hope you fixed your issue!

**Reply by jcard0na** [reply]
Thanks for the update, @WizOfWas 

Regarding your second issue, yes, it is a know problem that applies just to the Raspberry Pi 3 (#37).  The Pi 4 and 5 have smaller (micro) HDMI ports, which much bigger clearance to the mini-jack port.

**Comment by iwtwwc** [comment]
I really appreciate all the help from Haxophone creator and users. But I'm now stuck again unable to write to my SD card. If I hadn't been able to write to it once I'd suspect the card is bad. I really do have to be on the net BTW, the Imager just has [any] for raspberry pi device if I'm not online. I've tried using disk utility to first aid and erase, then Imager to reformat, then imager to load halo = failure. Get failure omitting any of the disk utility steps (I had to follow those steps to get it to work once). I'm saying no to all customization, I think I was also stuck in an endless reboot loop (green led never stops blinking) yesterday. Great tip about contact between the sound jack and Pi, but I'm trying a Zero W which isn't touching the jack.

**Comment by iwtwwc** [comment]
Yeah now both disk utility and pi imager are giving me the same error: (-69760) unable to write to last block of the device
Looks like I need a new 16 GB card, maybe this 32 GB was too much

**Reply by jcard0na** [reply]
Hi @iwtwwc 

My understanding is that you should be able to flash a smaller SD card image onto a bigger card.  Once booted, the haxophone would only "see" 16 GB and not the full size of the card.  But the image should work nonetheless.  
I have a big stash of 16 GB cards, and nothing else, so I cannot test this at the moment.  However, [other players have successfully used a 32 GB card](https://github.com/cardonabits/haxo-hw/discussions/59#discussioncomment-8035506).  That said, the card needs to be in good working condition.

**Comment by iwtwwc** [comment]
I'm starting to suspect my card is the issue. I've only had one successful write out of over a dozen attempts and I agree all the fooling with disk utility should not be needed. I did try another adaptor (microSD to my iMac's slot) but no luck. Guess I'll get a couple 16 GB disks and see if I can find a USB to SD reader, bouncing onto the old iMac is a hassle. Oh, I also tried writing the OS to the SD card that was at the top of the software list I get when online (the version from raspberry) and that failed too so it's not the haxo.img file :)

**Reply by jcard0na** [reply]
In the meantime I have...

1. Confirmed that you can flash the 16 GB image onto a bigger card.  I found a 128 GB card laying around, and it boots just fine.  
The disk size is reported as `100 GB` (instead of the advertised 128 GB, I suspect there are many bad sectors... that disk was sitting at the bottom of my drawer for ages...), but the partition size is `14.3 GB`, as in the original image.
```
sudo fdisk -l /dev/mmcblk0
Disk /dev/mmcblk0: 99.61 GiB, 106954752000 bytes, 208896000 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disklabel type: dos
Disk identifier: 0x3fc111cd

Device         Boot  Start      End  Sectors  Size Id Type
/dev/mmcblk0p1        8192   532479   524288  256M  c W95 FAT32 (LBA)
/dev/mmcblk0p2      532480 30533631 30001152 14.3G 83 Linux
```

2. Documented how to flash and get logs [here](https://github.com/cardonabits/haxo-hw/blob/main/docs/imaging.md)

Hope that it helps.

Cheers,

**Comment by iwtwwc** [comment, marked as answer]
Partial success! I bought an SD card reader (Insignia) and new SanDisk 32 GB (they had nothing smaller). Put the card in the reader right out of the package and installed haxo.img with the Imager No Problem! Put the card into Pi and about 1 minute later in made little sound!
I'm not 100%. Each key makes the same note (except top 4 = vibrato?) and at first it took blow+ press to get the 1 note but now just blowing makes the note. Octave shift up on the back works (on the one note). Middle key of left hand palm shifts note up (not an octave). Maybe I don't know how to play a saxophone (and I don't!!!)
Making progress!

**Reply by jcard0na** [reply]
Oh, that's great.  Here you can see all the fingerings:  https://haxo-notemap.nn.r.appspot.com/
Or here:  https://www.sebulli.com/haxo/notemap/
(both have the same info)

Start with the fingerings that require few fingers (like B4, A4, C4...) then gradually try to play the fingerings that require more fingers.  Let me know if there is any fingering that does not work, if any, and we'll try to troubleshoot.

Good progress!

**Comment by iwtwwc** [comment]
Oh awesome, I've arrived at the learning saxophone part of the Haxophone, thank you! Especially for your very fast and personal help in this forum, I can't wait to post a video of my first tune!
